dsl_parser/tests/test_inputs.py

```diff
@@ -810,3 +817,392 @@ def test_input_missing_attr_raises_keyerror(self):
+    b:
+        properties:
+            c:
+                type: c
+    c:
+        properties:
+            d:
+                type: integer
+
+"""
+        e = self.assertRaises(
+            DSLParsingException,
+            prepare_deployment_plan,
+            self.parse_1_2(yaml),
+            inputs={'some_input': {'e': {'c': {'d': 'should_be_int'}}}})
+        self.assertEqual(ERROR_INPUT_VIOLATES_DATA_TYPE_SCHEMA, e.err_code)
+        self.assertIn('Undefined property e', e.message)
+
+    def test_input_value_data_type_validation_raises_with_derived(self):
+        yaml = self.BASIC_VERSION_SECTION_DSL_1_2 + """
+inputs:
+    some_input:
+        type: b
+data_types:
+    a:
+        properties:
+            b:
+                type: integer
+    b:
+        derived_from: a
+        properties:
+            c:
+                type: integer
+"""
+        e = self.assertRaises(
+            DSLParsingException,
+            prepare_deployment_plan,
+            self.parse_1_2(yaml),
+            inputs={'some_input': {'c': 123}})
+        self.assertEqual(ERROR_INPUT_VIOLATES_DATA_TYPE_SCHEMA, e.err_code)
+        self.assertIn('is missing property b', e.message)
+
+    def test_input_default_value_type_validation_raises_with_derived(self):
+        yaml = self.BASIC_VERSION_SECTION_DSL_1_2 + """
+inputs:
+    some_input:
+        type: b
+        default:
+            c: 123
+data_types:
+    a:
+        properties:
+            b:
+                type: integer
+    b:
+        derived_from: a
+        properties:
+            c:
+                type: integer
+"""
+        ex = self._assert_dsl_parsing_exception_error_code(
+            yaml,
+            ERROR_MISSING_PROPERTY)
+        self.assertIn('is missing property b', ex.message)
+
+    def test_input_validate_type_raises(self):
+        yaml = self.BASIC_VERSION_SECTION_DSL_1_2 + """
+inputs:
+    some_input:
+        type: who_dis?
+"""
+        ex = self._assert_dsl_parsing_exception_error_code(
+            yaml,
+            ERROR_UNKNOWN_TYPE)
+        self.assertIn('Illegal type name', ex.message)
+
+    def test_input_validate_regex_successful(self):
+        yaml = self.BASIC_VERSION_SECTION_DSL_1_2 + """
+inputs:
+    some_input:
+        type: regex
+        default: '^$'
+"""
+        self.parse(yaml)
+
+    def test_input_validate_regex_raises(self):
+        yaml = self.BASIC_VERSION_SECTION_DSL_1_2 + """
+inputs:
+    some_input:
+        type: regex
+        default: '\\'
+"""
+        ex = self._assert_dsl_parsing_exception_error_code(
+            yaml,
+            ERROR_VALUE_DOES_NOT_MATCH_TYPE)
+        self.assertIn('Property type validation failed in', ex.message)
+        self.assertIn("type is 'regex'", ex.message)
+
+    def test_input_validate_list_successful(self):
+        yaml = self.BASIC_VERSION_SECTION_DSL_1_2 + """
+inputs:
+    some_input:
+        type: list
+        default: [one, two, 3]
+"""
+        self.parse(yaml)
```

Comment:
All the test that are just running parse only verifies it passes parse but don't verify that the values are there

Reply:
The code doesn't parse it any differently than not having `type` defined at all. This is just tests the type validation.

Reply:
Added type & default values checks.